antarest/study/business/aggregator_management.py

```diff
@@ -144,20 +225,98 @@ def _gather_all_files_to_consider(self) -> t.Sequence[Path]:
+            for area_or_link, files in filtered_files.items()
+            for file in files
+        ]
+        return all_output_files
+
+    def columns_filtering(self, df: pd.DataFrame, is_details: bool) -> pd.DataFrame:
+        # columns filtering
+        lower_case_columns = [c.lower() for c in self.columns_names]
+        if lower_case_columns:
+            if is_details:
+                filtered_columns = [CLUSTER_ID_COL, TIME_ID_COL] + [
+                    c for c in df.columns.tolist() if any(regex in c.lower() for regex in lower_case_columns)
+                ]
+            elif self.mc_root == MCRoot.MC_ALL:
+                filtered_columns = [
+                    c for c in df.columns.tolist() if any(regex in c.lower() for regex in lower_case_columns)
+                ]
+            else:
+                filtered_columns = [c for c in df.columns.tolist() if c.lower() in lower_case_columns]
+            df = df.loc[:, filtered_columns]
+        return df
+
+    def _process_df(self, file_path: Path, is_details: bool) -> pd.DataFrame:
+        if is_details:
+            un_normalized_df = self._parse_output_file(file_path, normalize_column_name=False)
+            df_len = len(un_normalized_df)
+            cluster_dummy_product_cols = sorted(
+                set([(x[CLUSTER_ID_COMPONENT], x[DUMMY_COMPONENT]) for x in un_normalized_df.columns])
+            )
+            actual_cols = sorted(set(un_normalized_df.columns.map(lambda x: x[ACTUAL_COLUMN_COMPONENT])))
+            new_obj: t.Dict[str, t.Any] = {k: [] for k in actual_cols}
+            new_obj[CLUSTER_ID_COL] = []
+            new_obj[TIME_ID_COL] = []
+            for cluster_id, dummy_component in cluster_dummy_product_cols:
+                for actual_col in actual_cols:
+                    col_values = un_normalized_df[(cluster_id, actual_col, dummy_component)].tolist()  # type: ignore
+                    new_obj[actual_col] += col_values
+                new_obj[CLUSTER_ID_COL] += [cluster_id for _ in range(df_len)]
+                new_obj[TIME_ID_COL] += list(range(1, df_len + 1))
+
+            prod_col = _infer_production_column(actual_cols)
+            if prod_col is not None:
+                new_obj[PRODUCTION_COLUMN_NAME] = new_obj.pop(prod_col)
+                actual_cols.remove(prod_col)
+
+            add_prod = [PRODUCTION_COLUMN_NAME] if prod_col is not None else []
+            columns_order = [CLUSTER_ID_COL, TIME_ID_COL] + add_prod + list(actual_cols)
+            df = pd.DataFrame(new_obj).reindex(columns=columns_order).sort_values(by=[TIME_ID_COL, CLUSTER_ID_COL])
+            df.index = pd.Index(list(range(1, len(df) + 1)))
+
+            return df
+
+        else:
+            return self._parse_output_file(file_path)
+
     def _build_dataframe(self, files: t.Sequence[Path], horizon: int) -> pd.DataFrame:
+        if self.mc_root not in [MCRoot.MC_IND, MCRoot.MC_ALL]:
```

Comment:
No because, a developer could modify a function without knowing anything about what happens in an other one ... Thus you should keep it or merge functions.

Reply:
In general I understand your point but here this private function is only used once so IMO it doesn't apply. If you follow this logic, you'll have to also do this check insie `_process_df` etc.